ccinput/tests/test_nwchem.py

```diff
@@ -0,0 +1,1430 @@
+from ccinput.tests.testing_utilities import InputTests
+from ccinput.exceptions import InvalidParameter, ImpossibleCalculation, UnimplementedError
+
+
+class NwchemTests(InputTests):
+
+    def test_sp_HF(self):
+        params = {
+            "nproc": 8,
+            "mem": "10000MB",
+            "type": "Single-Point Energy",
+            "file": "Cl.xyz",
+            "software": "nwchem",
+            "method": "HF",
+            "basis_set": "3-21G",
+            "charge": "-1",
+        }
+
+        inp = self.generate_calculation(**params)
+
+        REF = """
+        TITLE "File created by ccinput"
+        start Cl
+        memory total 10000 mb
+        charge -1
+
+        geometry units angstroms
+        Cl   0.00000000   0.00000000   0.00000000
+        end
+
+        basis
+        * library 3-21G
+        end
+
+        scf
+        singlet
+        end
+
+        task scf energy
+
+
+        """
+
+        self.assertTrue(self.is_equivalent(REF, inp.input_file))
+        
+    def test_sp_DFT(self):
+        params = {
+            "nproc": 8,
+            "mem": "10000MB",
+            "type": "Single-Point Energy",
+            "file": "Cl.xyz",
+            "software": "nwchem",
+            "method": "M06-2X",
+            "basis_set": "Def2-SVP",
+            "charge": "-1",
+        }
+
+        inp = self.generate_calculation(**params)
+
+        REF = """
+        TITLE "File created by ccinput"
+        start Cl
+        memory total 10000 mb
+        charge -1
+
+        geometry units angstroms
+        Cl   0.00000000   0.00000000   0.00000000
+        end
+
+        basis
+        * library Def2-SVP
+        end
+
+        dft
+        xc m06-2x
+        mult 1
+        end
+
+        task dft energy
+
+
+        """
+
+        self.assertTrue(self.is_equivalent(REF, inp.input_file))
+
+    def test_sp_DFT_specifications(self):
+        params = {
+            "nproc": 8,
+            "mem": "10000MB",
+            "type": "Single-Point Energy",
+            "file": "Cl.xyz",
+            "software": "nwchem",
+            "method": "M06-2X",
+            "basis_set": "Def2-SVP",
+            "charge": "-1",
+            "specifications": "scf(grid coarse)",
```

Comment:
I like this! Would `dft(grid coarse)` also work? An additional test for this would be great

Reply:
I will implement that in the changes of this commit.